### PR TITLE
BL-8752: Add load data from csv step in the pipeline

### PR DESCRIPTION
**Pull request by mrogowska**

### CODE REVIEW

File: jenkinsfile.groovy [hunk 1]
```diff
@@ -45,6 +45,11 @@ String manifestVersion = ""
 String assetsBasePath = ""
 String frontendArtifact = ""
 String backendArtifact = ""
+String documentsDir = "documents"
+String csvFileName = "RecallsFile.csv"
+String recallsTableName = "cvr-${params.environment}-recalls"
+String makesTableName = "cvr-${params.environment}-makes"
+String modelsTableName = "cvr-${params.environment}-models"
 net.sf.json.JSON manifestContent
```
Comment (ryszardm-kainos) [inline jenkinsfile.groovy:51]: With introduction of /common It should be added to manifest and checked in order to determine whether we should rebuild all lambdas(similar to MOTR)

File: jenkinsfile.groovy [hunk 2]
```diff
@@ -462,6 +491,40 @@ pipeline {
...
+              github.cvr_app.name,
+              globalValuesFactory.SSH_DEPLOY_GIT_CREDS_ID
+          )) {
+            failure("Failed to clone repository ${github.cvr_app.url}; branch: ${params.branch}")
+          }
+
+          dir ("${github.cvr_app.name}/${databaseScriptDir}") {
+            if (awsFunctionsFactory.awsCli("aws dynamodb update-table --region ${globalValuesFactory.AWS_REGION} --table-name cvr-${params.environment}-recalls --provisioned-throughput ReadCapacityUnits=500,WriteCapacityUnits=500")
+            .status) {
+              failure("Failure while increasing recalls table's read and write capasity")
```
Comment (ryszardm-kainos) [inline jenkinsfile.groovy]: capasity -> capacity

File: jenkinsfile.groovy [hunk 3]
```diff
@@ -462,6 +491,40 @@ pipeline {
...
+            }
+            if (sh (
+              script: "npm install && AWS_REGION=${globalValuesFactory.AWS_REGION} ENVIRONMENT=${params.environment} npm run loadData",
+              returnStatus: true
+            )) {
+              failure("Failed to load data to the database.")
+            }
+            if (awsFunctionsFactory.awsCli("aws dynamodb update-table --region ${globalValuesFactory.AWS_REGION} --table-name cvr-${params.environment}-recalls --provisioned-throughput ReadCapacityUnits=1,WriteCapacityUnits=1")
+            .status) {
+              failure("Failure while decreasing recalls table's read and write capasity")
```
Comment (ryszardm-kainos) [inline jenkinsfile.groovy]: capacity

File: jenkinsfile.groovy [hunk 4]
```diff
@@ -462,6 +491,40 @@ pipeline {
...
+              failure("Failure while increasing recalls table's read and write capasity")
+            }
+            if (sh (
+              script: "npm install && AWS_REGION=${globalValuesFactory.AWS_REGION} ENVIRONMENT=${params.environment} npm run loadData",
+              returnStatus: true
+            )) {
+              failure("Failed to load data to the database.")
+            }
+            if (awsFunctionsFactory.awsCli("aws dynamodb update-table --region ${globalValuesFactory.AWS_REGION} --table-name cvr-${params.environment}-recalls --provisioned-throughput ReadCapacityUnits=1,WriteCapacityUnits=1")
+            .status) {
```
Comment (ryszardm-kainos) [inline jenkinsfile.groovy]: This would typically be indented 2times (here and L510)

File: jenkinsfile.groovy [hunk 5]
```diff
@@ -462,6 +491,40 @@ pipeline {
       } //parallel
     } //stage deploy
 
+    stage ("Load data to DB") {
+      when  { expression { params.action == 'apply' }}
+      agent { node { label "${jenkinsCtrlNodeLabel} && ${account}" } }
```
Comment (ryszardm-kainos) [inline jenkinsfile.groovy]: not sure if `jenkinsCtrlNodeLabel` is appropriate for tasks involving `npm install`, I'd try using `jenkinsBuildLabel` node

File: jenkinsfile.groovy [hunk 6]
```diff
@@ -462,6 +491,40 @@ pipeline {
...
+            failure("Failed to clone repository ${github.cvr_app.url}; branch: ${params.branch}")
+          }
+
+          dir ("${github.cvr_app.name}/${databaseScriptDir}") {
+            if (awsFunctionsFactory.awsCli("aws dynamodb update-table --region ${globalValuesFactory.AWS_REGION} --table-name cvr-${params.environment}-recalls --provisioned-throughput ReadCapacityUnits=500,WriteCapacityUnits=500")
+            .status) {
+              failure("Failure while increasing recalls table's read and write capasity")
+            }
+            if (sh (
+              script: "npm install && AWS_REGION=${globalValuesFactory.AWS_REGION} ENVIRONMENT=${params.environment} npm run loadData",
```
Comment (ryszardm-kainos) [inline jenkinsfile.groovy]: Does this script do a full clear+load of data or does it do nothing when data is present? Maybe we could add a flag to skip this stage that someone could select if he is applying on top of an environment that is healthy? It wouldn't eat up our limits of rescaling dynamodb capacity

**Review by ryszardm-kainos — CHANGES_REQUESTED** [review]

File: jenkinsfile.groovy (hunk 1 above)
Comment (ryszardm-kainos) [reply jenkinsfile.groovy:51]: This is resolved

File: jenkinsfile.groovy (hunk 6 above)
Comment (mrogowska) [reply jenkinsfile.groovy]: It doesn't clear the data, I'll add a flag for skipping that stage - good idea

**Review by ryszardm-kainos — APPROVED** [review]

**Pull request merged**